File: ui/src/test/java/org/wcl/mfn/ui/utilities/NavigationBarChecker.java
package org.wcl.mfn.ui.utilities;

import static org.assertj.core.api.Assertions.*;
import org.jsoup.Jsoup;

public class NavigationBarChecker {
    private static final String HREF_ATTRIBUTE = "href";
    private static final String A_ATTRIBUTE = "a";

    private static final String HOME_TEXT = "Home";
    private static final String HOME_LINK = "/";

    private static final String CONTRACT_CALCULATOR_TEXT = "Contract Calculator";
    private static final String CONTRACT_CALCULATOR_LINK = "/contract-calculator";

    private static final String TOOLS_MENU_TEXT = "Tools";

    private static final int HOME_LINK_INDEX = 0;
    private static final int TOOLS_MENU_INDEX = 1;
    public static final int CONTRACT_CALCULATOR_INDEX = 2;

    private NavigationBarChecker() {}

    public static void checkNavigationBarContents(String htmlContent) {
        final var links = Jsoup.parse(htmlContent).select(A_ATTRIBUTE);

        assertThat(links.size()).isEqualTo(3);
        assertThat(links.get(HOME_LINK_INDEX).attr(HREF_ATTRIBUTE)).isEqualTo(HOME_LINK);
        assertThat(links.get(HOME_LINK_INDEX).text()).isEqualTo(HOME_TEXT);
        assertThat(links.get(TOOLS_MENU_INDEX).text()).isEqualTo(TOOLS_MENU_TEXT);
        assertThat(links.get(CONTRACT_CALCULATOR_INDEX).attr(HREF_ATTRIBUTE)).isEqualTo(CONTRACT_CALCULATOR_LINK);
        assertThat(links.get(CONTRACT_CALCULATOR_INDEX).text()).isEqualTo(CONTRACT_CALCULATOR_TEXT);
    }
}

File: ui/src/test/kotlin/org/wcl/mfn/ui/utilities/NavigationBarChecker.kt
package org.wcl.mfn.ui.utilities

import org.assertj.core.api.Assertions
import org.jsoup.Jsoup

object NavigationBarChecker {
    private const val HREF_ATTRIBUTE = "href"
    private const val A_ATTRIBUTE = "a"

    private const val HOME_TEXT = "Home"
    private const val HOME_LINK = "/"

    private const val CONTRACT_CALCULATOR_TEXT = "Contract Calculator"
    private const val CONTRACT_CALCULATOR_LINK = "/contract-calculator"

    private const val TOOLS_MENU_TEXT = "Tools"

    fun checkNavigationBarContents(htmlContent: String) {
        val links = Jsoup.parse(htmlContent).select(A_ATTRIBUTE)

        Assertions.assertThat(links.size).isEqualTo(3)
        Assertions.assertThat(links[0].attr(HREF_ATTRIBUTE)).isEqualTo(HOME_LINK)
        Assertions.assertThat(links[0].text()).isEqualTo(HOME_TEXT)
        Assertions.assertThat(links[1].text()).isEqualTo(TOOLS_MENU_TEXT)
        Assertions.assertThat(links[2].attr(HREF_ATTRIBUTE)).isEqualTo(CONTRACT_CALCULATOR_LINK)
        Assertions.assertThat(links[2].text()).isEqualTo(CONTRACT_CALCULATOR_TEXT)
    }
}
